pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -270,20 +270,42 @@ func (r *Reconciler) delete(ctx context.Context, instance *servingv1alpha1.Knati
 	if len(instance.GetFinalizers()) == 0 || instance.GetFinalizers()[0] != finalizerName {
 		return nil
 	}
+
 	logger.Info("Deleting resources")
-	var RBAC = mf.Any(mf.ByKind("Role"), mf.ByKind("ClusterRole"), mf.ByKind("RoleBinding"), mf.ByKind("ClusterRoleBinding"))
-	if len(r.servings) == 0 {
-		if err := r.config.Filter(mf.ByKind("Deployment")).Delete(); err != nil {
+
+	// List all KnativeServings to determine if cluster-scoped resources should be deleted.
+	kss, err := r.knativeServingClientSet.OperatorV1alpha1().KnativeServings("").List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all KnativeServings: %w", err)
+	}
+
+	// Only delete cluster-scoped resources if all KnativeServings are being deleted.
+	allBeingDeleted := true
+	for _, ks := range kss.Items {
+		if ks.GetDeletionTimestamp().IsZero() {
+			allBeingDeleted = false
+			break
+		}
+	}
+
+	if allBeingDeleted {
+		// If the list has only one element, we're currently reconciling the last one and thus
+		// can go ahead and delete cluster-scoped resources.
+		manifest, err := r.transform(ctx, instance)
```

Comment:
Along the line 263 before, you use r.transform(instance), but add ctx as another param here. Which one should be the correct one?

Reply:
I see. they are two different functions

Reply:
Both will eventually have a ctx once I land a followup of this that will use genreconciler for KnativeEventing as well.

It boils down to logger usage: We should always use the logger in the context, because it has all the... well... context already set to it.